dvc/stage/__init__.py

```diff
@@ -32,6 +32,35 @@
 INVALID_STAGENAME_CHARS = set(string.punctuation) - {"_", "-"}
 
 
+class ExternalOutputsError(DvcException):
+    pass
+
+
+def _check_no_externals(outs):
```

Comment:
I think, we should have this check outside on `add()` as it's not really a stage that's external but an output.

Reply:
@skshetry It is also used on `run` so it made sense to put it in one place. We also have some other out-relateted checks in create_stage already, like checking the cycles or output overlaps.

Reply:
Ahh, missed the `run`.

Reply:
Stage is already bloated, so I didn't want this check here. But, as it's also on `run`, it's okay, I guess.